crates/metaslang/bindings/src/lib.rs

```diff
@@ -525,28 +429,14 @@ impl<'a, KT: KindTypes + 'static> Reference<'a, KT> {
             .expect("Reference does not have a valid file descriptor")
     }
 
-    pub fn resolve_definition(&self) -> Result<Definition<'a, KT>, ResolutionError<'a, KT>> {
-        Resolver::build_for(self, ResolveOptions::Full).first()
-    }
-
     pub fn definitions(&self) -> Vec<Definition<'a, KT>> {
         Resolver::build_for(self, ResolveOptions::Full).all()
     }
 
-    pub(crate) fn non_recursive_resolve(
-        &self,
-    ) -> Result<Definition<'a, KT>, ResolutionError<'a, KT>> {
+    pub(crate) fn non_recursive_resolve(&self) -> Vec<Definition<'a, KT>> {
         // This was likely originated from a full resolution call, so cut
         // recursion here by restricting the resolution algorithm.
-        Resolver::build_for(self, ResolveOptions::NonRecursive).first()
-    }
-
-    pub(crate) fn has_tag(&self, tag: Tag) -> bool {
-        self.owner
-            .references_info
-            .get(&self.handle)
-            .and_then(|info| info.tag)
-            .is_some_and(|reference_tag| reference_tag == tag)
+        Resolver::build_for(self, ResolveOptions::NonRecursive).all()
```

Comment:
I wonder if the comment here needs to be updated? Not sure I'm following why the change to `all()`.

Reply:
Since we're no longer ranking the results, it makes no sense to pick the first result arbitrarily. This errs on the safe side and returns all results. This should not be a problem in the general case, since this non-recursive resolution should be applied to identifiers that are normally non-ambiguous (eg. contract names).